### PR TITLE
Fix parsing of mac address in SDDP Discovery

### PR DESCRIPTION
**Pull request by mlobstein**
Fix issue reported by: https://github.com/openhab/openhab-addons/issues/16964

If the SDDP host address field contains colons in the mac address, remove them.

@andrewfg @lolodomo

### CODE REVIEW

File: bundles/org.openhab.core.config.discovery.sddp/src/main/java/org/openhab/core/config/discovery/sddp/SddpDevice.java [hunk 1]
```diff
@@ -131,7 +131,7 @@ public SddpDevice(Map<String, String> headers, boolean offline) {
 
         String[] hostParts = host.split("-|_");
         macAddress = hostParts.length <= 1 ? ""
-                : hostParts[hostParts.length - 1].replaceAll("(..)(?!$)", "$1-").toLowerCase();
+                : hostParts[hostParts.length - 1].replaceAll(":", "").replaceAll("(..)(?!$)", "$1-").toLowerCase();
```
Comment (holgerfriedrich) [inline bundles/org.openhab.core.config.discovery.sddp/src/main/java/org/openhab/core/config/discovery/sddp/SddpDevice.java:134]:
```suggestion
                : hostParts[hostParts.length - 1].replace(":", "").replaceAll("(..)(?!$)", "$1-").toLowerCase();
```
No regex needed, ```replace``` should be more efficient.

**Review by holgerfriedrich — CHANGES_REQUESTED** [review]

**Review by andrewfg — CHANGES_REQUESTED** [review]
@mlobstein thanks for this. However since mac addresses can be delimited by either colon or dash characters, I would suggest to replace both of them as follows..

```
        macAddress = hostParts.length <= 1 ? ""
                : hostParts[hostParts.length - 1].replaceAll("[-:]", "").replaceAll("(..)(?!$)", "$1-").toLowerCase();
```

File: bundles/org.openhab.core.config.discovery.sddp/src/main/java/org/openhab/core/config/discovery/sddp/SddpDevice.java [hunk 2]
```diff
@@ -35,7 +35,7 @@ public class SddpDevice {
 
     /**
      * The host address of the device.
-     * For example: JVC_PROJECTOR-E0DADC152802
+     * For example: JVC_PROJECTOR-E0DADC152802 or JVC_PROJECTOR-E0:DA:DC:15:28:02
      * Note: the last 12 characters represent the MAC address of the device.
```
Comment (andrewfg) [inline bundles/org.openhab.core.config.discovery.sddp/src/main/java/org/openhab/core/config/discovery/sddp/SddpDevice.java:39]: .. the last 12 resp. 17 characters represent..

**Review by andrewfg — CHANGES_REQUESTED** [review]

**Comment by andrewfg** [comment]
> since mac addresses can be delimited by either colon or dash characters

HOWEVER the Host header is being split on underscore or dash characters, so a Mac address delimited on dashes would in fact fail for other reasons. So actually I would approve @holgerfriedrich 's proposed version. :)

**Comment by mlobstein** [comment]
Changes pushed, however the build is failing for unrelated reasons.

**Review by andrewfg — APPROVED** [review]
LGTM and thanks for updating the JUnit test too.

**Review by holgerfriedrich — APPROVED** [review]
LGTM, thanks!

**Comment by mlobstein** [comment]
@wborn @holgerfriedrich please merge before the final build of OH 4.2.0 if possible.

**Pull request merged**